### PR TITLE
Merge changes from mschnorb/VIC

### PR DESCRIPTION
**Pull request by mschnorb**
This request includes several feature updates, including 1) the addition of a second precipitation scaling factor (now scale precipitation separately for rain and snow), 2) addition of two output variables to capture changes in band elevation and band area (applicable during dynamic glacier simulations), and 3) an added variable to track glacier melt energy. Also included are corrections made to 1) the KIENZLE phase partitioning algorithm, 2) glacier energy balance calculation, and 3) the calculation of energy balance error.

### CODE REVIEW

File: GlacierEnergyBalance.c [hunk 1]
```diff
@@ -26,12 +25,13 @@ double GlacierEnergyBalance::calculate(double TSurf) {
   double Tmp;
   double VaporMassFlux;           /* Mass flux of water vapor to or from the intercepted snow (kg/m2s) */
   double Fbal;                    /* Energy balance at glacier surface */
+  double temp_IceDepth;			  /* Local variable to hold scaled value for IceDepth */
```
Comment (jameshiebert) [inline GlacierEnergyBalance.c]: You might consider making this `const`

File: SnowPackEnergyBalance.c [hunk 2]
```diff
@@ -90,16 +90,12 @@ double SnowPackEnergyBalance::calculate(double TSurf)
   /* Internal Routine Variables */
 
   double Density;                 /* Density of water/ice at TMean (kg/m3) */
-  /* double LongRadOut; */		  /* long wave radiation emitted by surface
-				     (W/m2) */
-  double NetRad;		  /* Net radiation exchange at surface 
-				     (W/m2) */
-  double RestTerm;		  /* Rest term in surface energy balance
-				     (W/m2) */
+  /* double LongRadOut; */		  /* long wave radiation emitted by surface (W/m2) */
```
Comment (jameshiebert) [inline SnowPackEnergyBalance.c]: Please remove commented out non-code.

File: StateIO.h [hunk 3]
```diff
@@ -468,8 +468,12 @@ enum StateMetaDataVariableIndices {
   ENERGY_GLACIER_FLUX, //new
   ENERGY_DELTACC_GLAC, //new
 
-	GLAC_MASS_BALANCE_INFO, // to store the Glacier Mass Balance information (grid cell num and equation) for use in VIC-RGM integration
-	NUM_GLAC_MASS_BALANCE_INFO_TERMS,
+  GLAC_MASS_BALANCE_INFO, // to store the Glacier Mass Balance information (grid cell num and equation) for use in VIC-RGM integration
+  NUM_GLAC_MASS_BALANCE_INFO_TERMS,
+
+  SOIL_PADJ_R, //new
```
Comment (jameshiebert) [inline StateIO.h]: Please don't put "//new" in the code. We have version control for a reason and can tell when things were added :)

File: calc_water_energy_balance_errors.c [hunk 4]
```diff
@@ -54,6 +54,7 @@ void calc_energy_balance_error(int    rec,
 			       double sensible,
 			       double grnd_flux,
 			       double snow_fluxes,
+				   double glac_fluxes,
```
Comment (jameshiebert) [inline calc_water_energy_balance_errors.c]: I think you have some whitespace mismatching here.

File: glacier_melt.c [hunk 5]
```diff
@@ -84,6 +84,7 @@ int glacier_melt(double Le,
     double *save_Qnet,
     double *save_advection,
     double *save_deltaCC_glac,
+	double *save_glacier_melt_energy,
```
Comment (jameshiebert) [inline glacier_melt.c]: ditto

File: initialize_atmos.c [hunk 6]
```diff
@@ -327,10 +327,9 @@ void initialize_atmos(atmos_data_struct        *atmos,
   }
 
   /* Scale precipitation using PADJ */
-
-  for (int idx = 0; idx < (state->global_param.nrecs*state->NF); idx++) {
+  /* for (int idx = 0; idx < (state->global_param.nrecs*state->NF); idx++) {
```
Comment (jameshiebert) [inline initialize_atmos.c]: Please remove commented out non-code.

File: surface_fluxes_glac.c [hunk 7]
```diff
@@ -238,10 +239,11 @@ int surface_fluxes_glac(
     Tair = atmos->air_temp[hidx] + soil_con->Tfactor[hru.bandIndex];
     step_prec[WET] = atmos->prec[hidx] / hru.mu * soil_con->Pfactor[hru.bandIndex];  // precipitation in mm
 
+    /** Calculate fraction of precipitation that falls as fain; scale rainfall and snowfall**/
```
Comment (mdfischer) [inline surface_fluxes_glac.c]: You probably meant to write "rain" instead of "fain"

**Comment by mschnorb** [comment]
This commit addresses outstanding recommended changes and comments. Specifically, temp_IceDepth converted to constant, non-code removed (including some additional occurrences), removed "//new" comments (plus additional occurrences), corrected white space mismatches (including additional occurrences) and corrected spelling errors.

File: surface_fluxes_glac.c [hunk 8]
```diff
@@ -281,6 +283,10 @@ int surface_fluxes_glac(
 
     LongUnderOut = step_energy.LongUnderOut;
 
+    /* Debugging */
+    /* if (rec == 1411)
+  	  fprintf(stderr, "Record %i\n", rec); */
+
     if (step_snow.swq > 0. || snowfall[WET] > 0.) {
```
Comment (jameshiebert) [inline surface_fluxes_glac.c:286]: This non-code comment still needs to be removed.

File: surface_fluxes_glac.c (hunk 7 above)
Comment (jameshiebert) [inline surface_fluxes_glac.c]: Still need to fix this typo.

**Comment by jameshiebert** [comment]
Round 2 looks much better. Once those last two fixes are made, I'm happy for you guys to merge this.

**Pull request merged**